### ISSUE TITLE
Support non-hex encoded signing keys

### ISSUE THREAD

**Issue by educosta** [Azure/azure-mobile-apps-node#432]
Hi,

I need generate a JWT in a C# code and use this to request api on azure mobile apps.
How can I do this?

Thanks

**Comment by danderson00** [comment]
Hi @educosta, have a look at the sample at https://github.com/Azure/azure-mobile-apps-node/tree/master/samples/custom-authentication, in particular api/login.js

**Comment by educosta** [comment]
Hi @danderson00,

I saw, but I need generate token in other place, not in mobile apps. Like this post https://shellmonger.com/2016/04/08/30-days-of-zumo-v2-azure-mobile-apps-day-5-custom-authentication/

But when I set X-ZUMO-AUTH with my generated token I getting 
```
{
  "name": "JsonWebTokenError",
  "message": "invalid signature"
}
```

**Comment by danderson00** [comment]
The JWT token must be signed with the same signing key as your web app. If you are using auth0, the article describes how to take the auth0 signing key and use it in Mobile Apps. If you are using another library, you need to obtain the signing key and set the website_auth_signing_key app setting.

**Comment by educosta** [comment]
Hi @danderson00 

Thank you for your response.

I'm using this code to generate token, I use a c# app to do this.

```
var payload = new Dictionary<string, object>()
                {
                    { "sub", userid },
                    { "aud", "urn:microsoft:windows-azure:zumo" },
                    { "iss", "urn:microsoft:windows-azure:zumo" },
                    { "exp", (int)DateTime.Now.AddMinutes(60).Subtract(utc0).TotalSeconds }
                };

                string token = JWT.JsonWebToken.Encode(payload, privateKey, JWT.JwtHashAlgorithm.HS256);
```

I'm using the same key in mobile apps, but not working.

**Comment by educosta** [comment]
I found a possible problem, in src/auth/index.js if I change 
`var key = hexStringToBuffer(configuration.secret);`
to 
`var key = configuration.secret;`

All works fine

**Comment by danderson00** [comment]
Ahh, good catch. This was added because signing keys generated by the Azure portal are always hex encoded strings. I'll add some detection and release this ASAP. Watch this space!

**Comment by danderson00** [comment]
@educosta Just released 2.1.6. Let me know if you have any issues!
